aten/src/ATen/cuda/detail/CUDAHooks.cpp

```diff
@@ -459,6 +462,132 @@ void CUDAHooks::deviceSynchronize(DeviceIndex device_index) const {
   c10::cuda::device_synchronize();
 }
 
+std::tuple<size_t, size_t, ptrdiff_t, std::string, std::string, std::string, uint64_t, bool>
+CUDAHooks::StorageShareDevice(const c10::Storage& storage) const {
+    // NOLINTNEXTLINE(cppcoreguidelines-init-variables)
+    auto shandle = c10::cuda::CUDACachingAllocator::shareIpcHandle(storage.mutable_data());
+    size_t ipc_memory_handle_size = shandle.handle.size();
+    size_t ipc_event_handle_size = sizeof(cudaIpcEventHandle_t);
+    std::string memory_handle(ipc_memory_handle_size, '\0');  // Using std::string instead of vector
+    std::string event_handle(ipc_event_handle_size, '\0');
+    std::string ref_counter(sizeof(std::string), '\0');
```

Comment:
I'm not sure yet why you're doing it this way but it seems strange

Reply:
@ezyang ref_counter must have enough space for memcpy, now: 
std::string ref_counter(sizeof(std::string), '\0');  ->  std::string ref_counter(sizeof(sent_data->handle()), '\0');